src/jdk.incubator.jextract/share/classes/jdk/incubator/jextract/Declaration.java

```diff
@@ -277,8 +292,8 @@
      * @param type the global variable declaration type.
      * @return a new global variable declaration with given name and type.
      */
-    static Declaration.Variable globalVariable(Position pos, String name, Type type) {
-        return new DeclarationImpl.VariableImpl(type, Declaration.Variable.Kind.GLOBAL, name, pos);
+    static Declaration.Variable globalVariable(Position pos, String name, Type type, Map<String, List<ConstantDesc>> attrs) {
```

Comment:
Uhmmm. I think I'd prefer not to have a Map accepting factory, but to have a `Declaration::withAttribute(String, ConstantDesc)` method. This makes the API a lot tidier (and is similar to what we did for layouts).

Reply:
I removed changes to factory methods, they are from parsing internally, so this is fine.

I agree withAttributes is preferred, but not certain about take single attribute/value. While this maybe the use case most of time, there are situations I need to do multiple modifications or copy and that leads to unnecessary rounds of object creation.

We can leave that out for now, since attributes are populated via parsing, and we don't have a use case where user should be adding attributes.